tests/test_vault.cairo

```diff
@@ -104,3 +110,84 @@ fn test_insufficient_balance_withdraw() {
     suite.vault.withdraw_liquidity(100);
     stop_cheat_caller_address(suite.vault.contract_address);
 }
+
+#[test]
+fn test_add_deposit_contract() {
+    let suite = setup_test_suite();
+    let user: ContractAddress = MOCK_USER.try_into().unwrap();
+    let deposit_address: ContractAddress = deploy_deposit_contract(user);
+    start_cheat_caller_address(suite.vault.contract_address, user);
+    suite.vault.add_deposit_contract(deposit_address);
+    // Check activeContracts
+    let activeContracts_after_adding = load(
+        suite.vault.contract_address,
+        map_entry_address(
+            selector!("activeContracts"), 
+            array![user.try_into().unwrap()].span()
+        ), 1,
+    );
+
+    stop_cheat_caller_address(suite.vault.contract_address);
+    assert(
+        (*activeContracts_after_adding[0]).try_into().unwrap() == deposit_address, 
+        'Deposit contract mismatch'
+    );
+}
+
+#[test]
+#[should_panic(expected: ('Deposit contract is zero',))]
+fn test_deposit_contract_address_is_zero() {
+    let suite = setup_test_suite();
+    let user: ContractAddress = MOCK_USER.try_into().unwrap();
+    let deposit_address: ContractAddress = 0.try_into().unwrap();
+    start_cheat_caller_address(suite.vault.contract_address, user);
+    suite.vault.add_deposit_contract(deposit_address);
+    stop_cheat_caller_address(suite.vault.contract_address);
+}
+
+#[test]
+#[should_panic(expected: ('Insufficient balance',))]
+fn test_protect_position_insufficient_balance() {
+    let suite = setup_test_suite();
+    let user: ContractAddress = HYPOTHETICAL_OWNER_ADDR.try_into().unwrap();
+    let user_2: ContractAddress = MOCK_USER.try_into().unwrap();
+    let user_amount: u256 = 100;
+
+    suite.token.transfer(user_2, user_amount); // Transfer tokens to caller
+    let deposit_address: ContractAddress = deploy_deposit_contract(user); // Deploy contract for owner
+    start_cheat_caller_address(suite.vault.contract_address, user);
+    suite.vault.protect_position(deposit_address, user_2, user_amount+50);
+    stop_cheat_caller_address(suite.vault.contract_address);
+}
+
+#[test]
+#[should_panic(expected: ('Caller must be owner or user',))]
+fn test_protect_position_caller_must_be_owner_or_user() {
+    let suite = setup_test_suite();
+    let user: ContractAddress = MOCK_USER.try_into().unwrap();
+    let user_2: ContractAddress = MOCK_USER_2.try_into().unwrap();
+    let user_amount: u256 = 100;
+
+    suite.token.transfer(user_2, user_amount); // Transfer tokens to caller
+    let deposit_address: ContractAddress = deploy_deposit_contract(user); // Deploy contract for owner
+    start_cheat_caller_address(suite.vault.contract_address, user);
+    suite.vault.protect_position(deposit_address, user_2, user_amount);
+    stop_cheat_caller_address(suite.vault.contract_address);
+}
+
+#[test]
+fn test_protect_position() {
```

Comment:
This test is failing, because: `user` has no tokens in `amounts` inside the contract and therefore nothing to protect position, so first, call `store_liquidity` for that user. Then it's going to fail inside of `protect_position` because `deposit_contract` has no position open, so call [loop_liquidity](https://github.com/djeck1432/spotnet/blob/main/src/deposit.cairo#L193) from the `Deposit` contract, use any token pair you want for that.